components/brave_wallet_ui/components/desktop/wallet-menus/wallet_settings_menu.tsx

```diff
@@ -86,7 +86,11 @@ export const WalletSettingsMenu = (props: Props) => {
     }
 
     const route =
-      selectedNetwork.coin === BraveWallet.CoinType.ETH ? 'ethereum' : 'solana'
+      selectedNetwork.coin === BraveWallet.CoinType.ETH
+        ? 'ethereum'
+        : selectedNetwork.coin === BraveWallet.CoinType.SOL
+          ? 'solana'
+          : 'cardano'
```

Comment:
nit. it looks like this should be a switch...

Reply:
Yeah, let's have a switch and fail an `assert` when a new type of dApp coin appears here